qtp_target_gene/validate.py

```diff
@@ -307,17 +308,20 @@ def _validate_demux_file(qclient, job_id, prep_info, out_dir, demux_fp,
         # Fix the sample ids
         # Do not modify the original demux file, copy it to a new location
         new_demux_fp = join(out_dir, basename(demux_fp))
-        copy(demux_fp, new_demux_fp)
-        # Need to catch an error
-        with open_file(new_demux_fp, 'r+') as f:
-            for old in f:
-                f.move(old, id_map[old])
-
-        # When we fix, we always generate the FASTQ and FASTA file
-        # By setting them to None, below will be generated
-        demux_fp = new_demux_fp
-        fastq_fp = None
-        fasta_fp = None
+        # this if is important so we don't regenerate the demux file if the
+        # user uploads fastq or fna
+        if demux_fp != new_demux_fp:
+            copy(demux_fp, new_demux_fp)
+            demux_fp = new_demux_fp
+
+        if bool(id_map):
```

Comment:
I don't think this 'if' is required, if this point in the code is reached, this will always evaluate to True.